### DISCUSSION TITLE
BM6

### DISCUSSION THREAD

**Discussion by eried** [KrystianD/bm2-battery-monitor#5] [General]
Hey! thanks for doing this work :) I just got a BM6 and it is so stupid that the whole communication is not open. Or their app is not on github. I guess the BM6 is just a BM2 with a temperature probe.

I wonder if you experimented on things like how long does the device store data? what is their "patented" SoC estimation method?

**Comment by KrystianD** [comment]
Hi @eried, I am really glad you like it!

When it comes to history - I haven't yet investigated this. I may work on this in the future, but for now I am fine with current value, as I am mostly interested in self-discharge monitoring.

Regarding SoC method - I never trust such Chinese "patented" algorithms, they are mostly designed to just show something, doesn't have to be accurate as most people do not have a way to verify its readings. It is plugged in parallel to battery terminals so it doesn't know anything about discharge or charge current, nothing about battery capacity, age, etc. I would use their SoC estimation with a dose of distrust :)

**Reply by eried** [reply]
Ah cool! I will poke around. I was having some issues with my battery and this device is very cool, but ofc I am afraid of the chinese app, 0 support, stop working one day (there is even a message of "not activated" if you dont have internet on the bm6)

**Reply by KrystianD** [reply]
Yeah, especially as their app sends a lot of information (including phone info, network, location!) to their servers.. :)

**Reply by KrystianD** [reply]
This is all that is being sent to their servers.. it should be stigmatized...

![image](https://user-images.githubusercontent.com/1083568/196968231-0528b4b0-b61b-4952-9e5c-e7b63f182304.png)
![image](https://user-images.githubusercontent.com/1083568/196968488-712fbab2-29ad-42e6-a7b4-fcd04a3670c8.png)
![image](https://user-images.githubusercontent.com/1083568/196968660-179e9ce2-d4e0-456b-a5af-75c790d16b00.png)

**Reply by eried** [reply]
lol, it would be easy to spam that endpoint, but that will just make a more obfuscated new firmware. Do you have any luck recompiling their app? it has a problem in the graphs which I would like to fix, or it could even mean an open version of the app without those reports...

**Reply by KrystianD** [reply]
Nope, I didn't even try (I never intended to use it, uninstalled this garbage as soon as I grabbed DEXes). It contains tons of third party dependencies (ads, ads, ads and even more tracking and reporting) and lot of parts were not decompiled correctly.

But you probably can take quite significant amount of code (mostly for connection handling) and create a custom app.
